modules/auxiliary/gather/impersonate_ssl.rb

```diff
@@ -42,8 +43,26 @@ def initialize(info={})
       ])
   end
 
+  def getCert(rhost,rport,sni)
+      ctx = OpenSSL::SSL::SSLContext.new
+      sock = TCPSocket.new(rhost, rport)
+      ssl = OpenSSL::SSL::SSLSocket.new(sock, ctx)
+      if sni
+        ssl.hostname = sni
+      end
+      ssl.connect
+      cert = ssl.peer_cert
+      return cert
+  end
```

Comment:
So this, whilst nice, will break when pivoting comes into play. We should really be using the Rex socket methods and you can see that `peer_cert` is contained in https://github.com/rapid7/rex-socket/blob/86d71342aa8957fdc88f0075213b1cad8591ff92/lib/rex/socket/ssl_tcp.rb#L331-L333